src/react/components/pages/projectSettings/projectMetrics.tsx

```diff
@@ -0,0 +1,260 @@
+                        </span>
+                    </li>
+                    <li className="list-group-item">
+                        <b>Number of assets visited: </b>
+                        <span
+                            className="visited-asset-count badge badge-light badge-pill">
+                            {this.getVisitedAssetsCount()}
+                        </span>
+                    </li>
+                    <li className="list-group-item">
+                        <b>Total tagged assets: </b>
+                        <span
+                            className="tagged-asset-count badge badge-light badge-pill">
+                            {this.getTaggedAssetCount()}
+                        </span>
+                    </li>
+                    <li className="list-group-item">
+                        <b>Number of regions: </b>
+                        <span className="regions-count badge badge-light badge-pill">
+                            {this.getRegionsCount()}
+                        </span>
+                    </li>
+                    <li className="list-group-item">
+                        <h5>Total tag categories:
+                            <span
+                                className="tag-categories-count badge badge-light badge-pill">
+                                {this.getTagCategoriesCount()}
+                            </span>
+                        </h5>
+                        <ul>{renderTagCategories}</ul>
+                    </li>
+                    <li className="list-group-item">
+                        <h5>Per tag totals (number of instances of that tag) </h5>
+                        <span className="tags-map">
+                            <ul>{renderTagCount}</ul>
+                        </span>
+                    </li>
+                    <li className="list-group-item">
+                        <b>Average tags per asset: </b>
+                        <span
+                            className="average-tags-count badge badge-light badge-pill">
+                            {this.getAverageTagsCount()}
+                        </span>
+                    </li>
+                </ul>
+            </div>
+        );
+    }
+
+    private async getAssetsAndMetadata() {
+        const assetService = new AssetService(this.props.project);
+        const sourceAssets = await assetService.getAssets();
+
+        const assetsMap = this.props.project.assets;
+        let projectAssetsMetadata = [];
+        if (assetsMap) {
+            const assets = _.values(assetsMap);
+            const temp = assets.map((a) => assetService.getAssetMetadata(a));
+            projectAssetsMetadata = await Promise.all(temp);
+        }
+
+        this.setState({
+            loading: false,
+            sourceAssets,
+            projectAssetsMetadata,
+        });
+    }
+
+    /**
+     * Count the number of tagged images or video frames
+     */
+    private getTaggedAssetCount() {
```

Comment:
Nit: Might be easier to test if these functions were just public methods on a static helper class. Just a thought, I'll leave it up to you